hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,83 +10,145 @@ This module conforms to the [Comportable guideline for peripheral functionality.
-
-It is assumed that any entropy source will drop bits.
-This means that the FIFO used to collect up the entropy bits will fill, and until the firmware or hardware starts pulling out bits, continuously generated entropy source bits may be dropped from usage.
-The process of filling and draining the FIFOs is the same, independent of what speed the entropy generation rate is at.
-
-Once the `es_entropy_valid` status bit is set, the firmware will read the {{< regref "ES_ENTROPY" >}} register as many times as needed to get the required entropy bits, typically 128 bits or 256 bits at a time.
-For ease of firmware operation, the status bit does not need to be reset.
-A read of the {{< regref "ES_THRESH" >}} register could be done to verify how many entries are in the FIFO at the current time.
-The {{< regref "ES_THRESH" >}} register will determine when the status bit is set, or when the interrupt will assert if enabled.
-For this revision, this same {{< regref "ES_THRESH" >}} register has a range of between 1 and 4 entries in the TLUL FIFO.
-
-For hardware consumption of entropy, if the HWIF FIFO is not empty, and the entropy consuming hardware is ready, then entropy bits will be transferred over the hardware entropy interface.
-
-Health tests, when enabled in the {{< regref "ES_CONF" >}} register, will run continuously.
-Two tests are implemented in hardware, the repetition count test (RCT), and the adaptive proportion test (APT).
-
-There are four health test blocks, one per inbound rng bus bit.
-Each health test block works independently from one another.
-The checks are done on the serial stream input, and an optional feature allows for passing entropy from a single bit stream source.
-Once the streams have passed the health tests, it becomes available for downstream consumption.
-If the health tests ever fail, the ES FIFO is prevented from receiving additional entropy bits up.
-The entropy health tests will continue until a clean stream is detected, at which time the ES FIFO will start collecting bits again.
-The {{< regref "ES_RCT_HEALTH" >}} register and the  {{< regref "ES_APT_HEALTH" >}} register set the cutoff limits for the health tests.
-
-The above process will be repeated for as long as entropy bits are to be collected.
-At any time, the enable bit can be cleared, and the entropy generation will halt immediately.
 
+After a reset, the ENTROPY_SRC block will start up in boot-time mode by default.
+This feature is designed to provide an initial seed's worth of entropy with lower latency than the normal FIPS/CC compliant health check process.
+Health testing will still be performed on boot-time mode entropy, but the window of checking is, by default, 384 bits instead of 1024 bits.
+When entropy is delivered to the downstream hardware block, a signal will indicate what type of entropy it is - FIPS compliant or not.
+Boot-time mode can be completely disabled in the {{< regref "CONF" >}} register.
```

Comment:
As a practical matter, if this register is not preserved through reset cycles, I don't think there's any way for firmware to actually set this disable bit, as the firmware hasn't even been fetched yet at the time the boot entropy is being extracted to configure the memory scrambling... right?

Reply:
After every reset, a single FW right is required to enable ES, so it is possible to set this bit in time. However, it is essentially a chicken switch and is not expected to be used.

Reply:
Ah, my point was that there might be (I'm not sure) a bit of a chicken and egg thing, unless we are running directly out of ROM.  It's not so much an issue for you as for the large group of people who might be looking at these docs and thinking about the boot timeline/sequence.